runelite-client/src/main/java/net/runelite/client/plugins/combatlevel/CombatLevelPlugin.java

```diff
@@ -112,4 +136,127 @@ public void onGameTick(GameTick event)
 
 		combatLevelWidget.setText("Combat Lvl: " + DECIMAL_FORMAT.format(combatLevelPrecise));
 	}
+
+	@Subscribe
+	public void onExperienceChanged(ExperienceChanged event)
+	{
+		final Skill skill = event.getSkill();
+
+		if (!COMBAT_SKILLS.contains(skill))
+		{
+			return;
+		}
+
+		levelsUntilTooltip = null;
+
+		final int level = Math.min(Experience.MAX_REAL_LEVEL, Experience.getLevelForXp(client.getSkillExperience(skill)));
+		cmbSkillLevels.put(skill, level);
+	}
+
+	Tooltip getLevelsUntilTooltip()
+	{
+		if (levelsUntilTooltip != null)
+		{
+			return levelsUntilTooltip;
```

Comment:
check for == null and build here and always return `levelsUntilTooltip` under the if statement

Reply:
Good point, fixed